### PR TITLE
Be more user-friendly with non-existing user v3

### PR DESCRIPTION
**Pull request by lukashino**
Follow-up of https://github.com/OISF/suricata/pull/9657

Redmine ticket:

https://redmine.openinfosecfoundation.org/issues/6278

Describe changes:
- edit FatalError message

To use a pull request use a branch name like pr/N where N is the
pull request number.

Alternatively, SV_BRANCH may also be a link to an
OISF/suricata-verify pull-request.

SV_REPO=
SV_BRANCH=pr/1440
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

**Review by jasonish — APPROVED** [review]

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16313

File: src/suricata.c [hunk 1]
```diff
@@ -2157,16 +2157,14 @@ static int InitRunAs(SCInstance *suri)
     if (suri->do_setuid == TRUE) {
         if (SCGetUserID(suri->user_name, suri->group_name,
```
Comment (victorjulien) [inline src/suricata.c:2158]: it looks like SCGetUserID can't return anything else than 0?

File: src/suricata.c [hunk 2]
```diff
@@ -2157,16 +2157,14 @@ static int InitRunAs(SCInstance *suri)
...
                         &suri->userid, &suri->groupid) != 0) {
-            SCLogError("failed in getting user ID");
-            return TM_ECODE_FAILED;
+            FatalError("failed to get user ID");
         }
 
         sc_set_caps = TRUE;
     /* Get the suricata group ID to given group ID */
     } else if (suri->do_setgid == TRUE) {
         if (SCGetGroupID(suri->group_name, &suri->groupid) != 0) {
```
Comment (victorjulien) [inline src/suricata.c:2166]: same here

**Review by victorjulien — CHANGES_REQUESTED** [review]
looks like we have some dead code

**Comment by lukashino** [comment]
Dead code removed in https://github.com/OISF/suricata/pull/9722

**Pull request closed without merging**